__tests__/unit/fact.test.js

```diff
@@ -0,0 +1,415 @@
+        text:
+          "Stephen I becomes King of Hungary, which is established as a Christian kingdom",
+      },
+    });
+    expect(sentence).toBe(
+      "1000 is the year that Stephen I becomes King of Hungary, which is established as a Christian kingdom on December 25th."
+    );
+  });
+
+  test("Return sentence when year is invalid", function () {
+    const sentence = getSentence({
+      wantFragment: undefined,
+      type: "year",
+      number: 10000000000,
+      data: {
+        text: "nothing remarkable happened",
+      },
+    });
+    expect(sentence).toBe(
+      "10000000000 is the year that nothing remarkable happened."
+    );
+  });
+});
+
+describe("getSentence() for types: trivia and math", () => {
+  // trivia and math share the same prefix
+  test("When wantFragment is true, returns partial text only", function () {
+    const sentence = getSentence({
+      wantFragment: { fragment: true },
+      number: 88,
+      type: "trivia",
+      data: {
+        text: "the number of keys on a piano (36 black and 52 white)",
+      },
+    });
+    expect(sentence).toBe(
+      "the number of keys on a piano (36 black and 52 white)"
+    );
+  });
+
+  test("When wantFragment is undefined, returns full sentence", function () {
+    const sentence = getSentence({
+      wantFragment: undefined,
+      number: 88,
+      type: "trivia",
+      data: {
+        text: "the number of keys on a piano (36 black and 52 white)",
+      },
+    });
+    expect(sentence).toBe(
+      "88 is the number of keys on a piano (36 black and 52 white)."
+    );
+  });
+
+  test("When trivia number is invalid, returns sentence", function () {
+    const sentence = getSentence({
+      wantFragment: undefined,
+      number: 88888,
+      type: "trivia",
+      data: {
+        text: "an unremarkable number",
+      },
+    });
+    expect(sentence).toBe("88888 is an unremarkable number.");
+  });
+});
+
+describe("getSentence() for type 'date' ", function () {
+  test("When wantFragment is undefined, returns full sentence", function () {
+    const sentence = getSentence({
+      wantFragment: undefined,
+      number: 1 / 1,
+      type: "date",
+      data: {
+        text: "that Victoria is crowned princess of Sweden",
+        year: 1980,
+      },
+    });
+    expect(sentence).toBe(
+      "January 1st is the day in 1980 that that Victoria is crowned princess of Sweden."
+    );
+  });
+});
+
+describe("getDefaultMsg() for all 4 types", function () {
+  test("random default msg for type:'year'", function () {
+    const yearMsgs = [
+      "9999999999 is the year that nothing remarkable happened.",
+      "9999999999 is the year that the Earth probably went around the Sun.",
+      "9999999999 is the year that nothing interesting came to pass.",
+      "9999999999 is the year that we do not know what happened.",
+    ];
+    const sentence = getDefaultMsg({
+      number: 9999999999,
+      type: "year",
+      options: {},
```

Comment:
would be nice if you could just not provide `options` at all if you didn't want to pass any.

Reply:
@mmmaaatttttt when `options : { }` is removed, `TypeError: Cannot read property 'fragment' of undefined` occurs. 

The return statement is looking for an `options` object: 
```
return getSentence({
    wantFragment: options.fragment,
    number: number,
    type: type,
    data: data,
  });
```

I'm leaning towards keeping the `options : { }` in the tests ?

Reply:
Or another option would be to modify `function getDefaultMsg({ number, type, options })` to `function getDefaultMsg({ number, type, options = {} })` ?

Reply:
Sorry for the spam, I think I'll go with options b and modify the getDefaultMsg() to have a default empty object as options if no options is passed in